app/assets/javascripts/discourse/app/models/topic.js

```diff
@@ -461,6 +462,23 @@ export default class Topic extends RestModel {
     return visible !== undefined ? !visible : undefined;
   }
 
+  @discourseComputed("visibility_reason_id")
+  visibilityReasonTranslated() {
+    if (
+      this.visibility_reason_id &&
+      this.visibility_reason_id !== 99 // unknown
+    ) {
+      const reasonKey = Object.keys(TOPIC_VISIBILITY_REASONS)[
```

Comment:
This is getting key by value? Maybe this could work?

```javascript
const reasonKey = Object.keys(TOPIC_VISIBILITY_REASONS).find(key => TOPIC_VISIBILITY_REASONS[key] === value)
```

Reply:
Ah yes this works, nice!